### DISCUSSION TITLE
Lesson 14:  Box contract Forge Build Issue: No arguments passed to the base constructor error

### DISCUSSION THREAD

**Discussion by danstam** [Cyfrin/foundry-full-course-cu#1332] [Q&A] [answered]
I've noticed that the Box contract only compiles for me when I include the 'abstract' keyword. This makes sense to me since the Ownable.sol contract from OpenZeppelin has a constructor.

However, I'm a bit puzzled because in the course material, the Box contract is labeled as just "contract" without the 'abstract' keyword, and it's still compiling. I must be missing something here. Any insights on why this might be the case in the course, or if there's a specific reason for this difference

The error without "abstract" for reference:
```
[⠒] Compiling...
[⠢] Compiling 1 files with 0.8.21
[⠆] Solc 0.8.21 finished in 6.23ms
Error: 
Compiler run failed:
Error (3415): No arguments passed to the base constructor. Specify the arguments or mark "Box" as abstract.
 --> src/Box.sol:6:1:
  |
6 | contract Box is Ownable {
  | ^ (Relevant source part starts here and spans across multiple lines).
Note: Base constructor parameters:
  --> lib/openzeppelin-contracts/contracts/access/Ownable.sol:38:16:
   |
38 |     constructor(address initialOwner) {
   |                ^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by 0xusmanf** [comment, marked as answer]
This is happening because you are using the latest version of OpenZeppelin contracts, in the latest version you have to specify the owner address in the constructor.

In the old version used in the tutorial, it automatically sets the `msg.sender` as the owner and does not require you to pass the owner address to the constructor.

**Reply by danstam** [reply]
Thank you👍

**Comment by YourDailyBlockchain** [comment]
I passed the owner as msg.sender  this way and it compiled:
contract Box is Ownable(msg.sender)
